src/sonic-yang-models/yang-models/sonic-device_metadata.yang

```diff
@@ -250,6 +250,13 @@ module sonic-device_metadata {
                                  created, regardless of the CONFIG_DB buffers configuration.";
                 }
 
+                leaf supporting_bulk_counter_groups {
+                    type string;
+                    description "This field contains a list of counter groups that support bulk operation.
```

Comment:
Could you add the yang constrains for "This field contains a list of counter groups"? Is it true counter groups are also in ConfigDB and have yang models?

Reply:
@qiluo-msft @ganglyu 
Thanks for the comments but it's different.
Unlike other tables, the instance of counter-groups is defined as `container` instead of `list`.

Generic table definition. all the instances are defined in `list BUFFER_PROFILE_LIST`
**For this kind of definition, we can define an item which references the instances as `leaf-list` with type `leafref`.**

```
    container sonic-buffer-profile {

        container BUFFER_PROFILE {

            list BUFFER_PROFILE_LIST {
                key "name";

                leaf name {
                    type string;
                    description "Buffer Profile name";
                }

                leaf static_th {
                    type uint64;
                    description "The maximum size of the buffer pool the port can occupy";
                }

                leaf dynamic_th {
                    type int32 {
                        range "-8..7";
                    }
                    description "The maximum proportion of the free size of the buffer pool the port can occupy (Alpha value)";
                }

...
            }
        }
    }
```

However, the flex counter table is defined as the following.
**All the instances are explicitly enumerated as `container`s instead of `leaf-list`. In this sense, I do not find a way to define the items that reference flex counter table instances, like `supporting_bulk_counter_groups `, as a `leaf-list` using `leafref` type.
Neither can we add any constrains here.**

```
    container sonic-flex_counter {

        container FLEX_COUNTER_TABLE {

            /* typedef specific to FLEX_COUNTER_TABLE */
            typedef flex_status {
                type enumeration {
                    enum enable;
                    enum disable;
                }
            }

            typedef flex_delay_status {
                type stypes:boolean_type;
            }

            typedef poll_interval {
                type uint32 {
                    range 100..4294967295;
                }
            }

            typedef bulk_chunk_size {
                type uint32 {
                    range 1..4294967295;
                }
            }

            description "FLEX_COUNTER_TABLE part of config_db.json";

            /* below are in alphabetical order */

            container BUFFER_POOL_WATERMARK {
                /* BUFFER_POOL_WATERMARK_STAT_COUNTER_FLEX_COUNTER_GROUP */
                leaf FLEX_COUNTER_STATUS {
                    type flex_status;
                }
                leaf FLEX_COUNTER_DELAY_STATUS {
                    type flex_delay_status;
                }
                leaf POLL_INTERVAL {
                    type poll_interval;
                }
            }

            container QUEUE_WATERMARK {
...
            }
        }
    }
```